### PR TITLE
A Set of Legendairy New Drinks!

### PR DESCRIPTION
**Pull request by MintyPhresh**
## About this PR
This PR is a set of 9 new drink recipes based around "power cola," a mixture of condensed cola syrup, cola, and a healthy splash of plasma. The goal for each of these drinks is for them each to have some kind of mechanical twist, rather than just being interchangeable for any other drink with the same alcohol content. 

### Recipes
**Cola:** (1) Cola Syrup + (5) Soda Water -> (6) Cola

**Cola Syrup:** (4) Cola + (1) VHFCS + (1) Plasma @ 360K -> (2) Cola Syrup

**Power Cola:** (1) Cola + (1) Cola Syrup -> (1) Power Cola

### Drinks
**Legendairy**
(1) Power Cola + (1) Super Milk -> (2) Legendairy
ABV 0% - When it's in your system, has a chance to force you to pick up / drink from drinking glasses that have some amount of Legendairy in them. Funny blabs about what a great drink it is... Until you exceed 40 units in your body, after which the blabs are entirely negative about the drink until you get rid of it.

**Colamara Chaos**
(1) Power Cola + (0) Ice + (1) Liquid Dark Matter + (1) Bitters -> (3) Colamara Chaos + (0) Ice
ABV 10% - Special RP motive effects. See table below.

**Roaring Waters**
(1) Power Cola + (0) Ice + (5) Tonic Water + (5) Soda Water -> (10) Roaring Waters + (0) Ice
ABV 0% - Special RP motive effects. See table below.

**Redspot**
(1) Power Cola + (1) Cherry Juice + (1) Milk -> (3) Redspot
ABV 0% - Makes light showers on mixing; Special RP motive effects. See table below.

**Rotorua Ravager**
(1) Power Cola + (1) Steam + (1) Lipolicide + (1) Ethanol -> (4) Rotorua Ravager
ABV 40% - Gets stronger as you run out of stamina-- to a maximum of 400% ABV (a little stronger than Murdini) at 40% stamina.

**Vampire**
(1) Power Cola + (1) Salt + (1) Vodka + (1) Blood -> (4) Vampire
ABV 25% - Drains 4u blood per tick. Gets stronger as you lose blood, to a maximum of 600% ABV at 200u.

**Wipeout**
(1) Power Cola + (1) Plasma + (1) Espresso + (1) Bourbon + (1) Simple Syrup + (1) Prairie Fire -> (6) Wipeout
ABV ??% - 10% chance to decay into 1 unit of bourbon. 15% chance to decay into one unit of espresso. 21% chance to decay into 1 units of VHFCS. 5% chance to decay into 0.4 units of Prairie Fire. Designed to obliterate Slow Metabolism users, who get ~8x as many chems from this as a normal person.

**The Lost Treasure of Kalimero**
(1) Power Cola + (1) Port -> (2) The Lost Treasure of Kalimero
ABV 80% - Can only be made in cocktail shakers, must be pure. Teleports the container away to a random unlocked crate/locker on-station when mixed. Great taste!

```
Comparison of optimal standard bar orders 
with selected drinks from this PR

                 Restore   Decay     Thirst per 
Drink          | rate    | rate    | unit volume
------------------------------------------------
Triple C       | 2         0.4       5
6 fruit juices | 9         2.4       3.75

Colamara Chaos | 13        3         5.3
Roaring Waters | 0.5       0.05      10
Redspot        | 1.2       0.2       6
Lost Treasure  | 15        1         15
```
## Why's this needed?
Unlike chemistry, which has a couple useful precursor chems they can make, bartending doesn't really have any. (The closest might be Simple Syrup, which is frankly easier to make on-demand now that the cocktail shaker fits in chem heaters.) These drinks are intended to be a multi-stage process that can be made easier with some forethought, and are intended to be well worth the extra effort (though I worry they may be too weak at the moment).

```changelog
(u)mintyphresh
(*)Added 9 new drinks to the game. Recipes in the minor changes tab.
(+)
(4) Cola + (1) VHFCS + (1) Plasma @ 360K -> (2) Cola Syrup
(1) Cola + (1) Cola Syrup -> (1) Power Cola
(1) Power Cola + (1) Super Milk -> (2) Legendairy
(1) Power Cola + (0) Ice + (1) Liquid Dark Matter + (1) Bitters -> (3) Colamara Chaos + (0) Ice
(1) Power Cola + (0) Ice + (5) Tonic Water + (5) Soda Water -> (10) Roaring Waters + (0) Ice
(1) Power Cola + (1) Cherry Juice + (1) Milk -> (3) Redspot
(1) Power Cola + (1) Steam + (1) Lipolicide + (1) Ethanol -> (4) Rotorua Ravager
(1) Power Cola + (1) Salt + (1) Vodka + (1) Blood -> (4) Vampire
(1) Power Cola + (1) Plasma + (1) Espresso + (1) Bourbon + (1) Simple Syrup + (1) Prairie Fire -> (6) Wipeout
(1) Power Cola + (1) Port -> (2) The Lost Treasure of Kalimero (Must be pure, and in cocktail shaker)
```

### CODE REVIEW

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -1993,6 +1993,153 @@
...
+		result_amount = 2
+		mix_phrase = null
+		mix_sound = FALSE
+		drinkrecipe = TRUE
+
+		does_react(var/datum/reagents/holder)
+			var/atom/my_atom = holder.my_atom
+
+			if(length(holder.reagent_list) > length(src.required_reagents)) // drink must be pure to mix-- no teleporting chem bombs
+				return FALSE
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Chemistry-Recipes.dm:2062]:
```suggestion
```
Okay but teleporting chem bombs randomly does sound funny and not all that dangerous. Blame me if we have to nerf this later :)

File: code/modules/chemistry/Chemistry-Tools.dm [hunk 2]
```diff
@@ -118,7 +118,7 @@ ABSTRACT_TYPE(/obj/item/reagent_containers)
 		if (!istype(src, /obj/item/reagent_containers/glass) && !istype(src, /obj/item/reagent_containers/food/drinks))
 			return ..()
 
-		if (usr.stat || usr.getStatusDuration("knockdown") || BOUNDS_DIST(usr, src) > 0 || BOUNDS_DIST(usr, over_object) > 0)  //why has this bug been in since i joined goonstation and nobody even looked here yet wtf -ZeWaka
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Chemistry-Tools.dm:121]: Almost none of the changes in this file appear to be intentional, you're reverting several other PRs.

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 3]
```diff
@@ -119,6 +119,82 @@ datum
...
+							"The vile taste of [coke] [milk] lingers in your throat.",\
+							"You feel nauseous. You resolve to never drink [coke] [milk] again.")))
+						else if (probmult(6))
+							M.say(pick("I don't want any more of that [coke] [milk].",\
+								"I'm not thirsty anymore.",\
+								"I want to go home.",\
+								"I'm tired.",\
+								"I feel sick.",\
+								"[get_area(M)] looks too exciting for me.",\
+								"I want to be on a station more thrilling than [the_station_name]."))
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Reagents-FoodDrink.dm:194]: Not a fan of drinks that make you say stuff constantly while they're in you (bojaaaacks). Having it make you say something when you compulsively grab a glass and drink from it is funny but the constant "coge milg" spam will get old fast.

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 4]
```diff
@@ -1993,6 +1993,153 @@
...
+				var/obj/my_obj = my_atom
+				my_obj.set_loc(get_a_random_station_unlocked_container_with_no_others_on_the_turf())
+
+			if(my_atom.loc != mixperson) //checks if the drink actually teleported
+				mix_phrase = "The mixture teleports away! Go find it!"
+			else
+				mix_phrase = "The mixture groans scarily, turning in on itself and disintegrating."
+				holder.remove_reagent("lostcoke", created_volume)
+			for(var/mob/living/M in AIviewers(7, get_turf(my_atom)) )
+				boutput(M, SPAN_NOTICE("[bicon(my_atom)] [mix_phrase]"))
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Chemistry-Recipes.dm:2084]: Why is this here?

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
I like these, more cool bartender drinks are always good!

I think I said this before, but why does cola syrup exist here? It's easily confused with simple syrup and only used to make power cola. Why not just make that recipe produce power cola instead of adding an extra step/chem for no reason?

File: code/modules/chemistry/Chemistry-Recipes.dm (hunk 1 above)
Comment (MintyPhresh) [reply code/modules/chemistry/Chemistry-Recipes.dm:2062]: I can logically reason out that making this possible would be more interesting for the game as a whole, but I don't really *want* this drink to be used as a teleporting chem bomb... it feels out of flavor. You could use chems like chocolate and smoke powder to trivialize the search process....

**Comment by MintyPhresh** [comment]
> I think I said this before, but why does cola syrup exist here? It's easily confused with simple syrup and only used to make power cola. Why not just make that recipe produce power cola instead of adding an extra step/chem for no reason?
> 
Many moons ago, this PR was inspired by Super Milk, which has a near-identical synthesis path and theming. I wanted to keep the parallels between the two there.

> Not a fan of drinks that make you say stuff constantly while they're in you (bojaaaacks). Having it make you say something when you compulsively grab a glass and drink from it is funny but the constant "coge milg" spam will get old fast.
> 
You're right. Rework pending

**Comment by MintyPhresh** [comment]
Closing this PR for now, as I don't really feel comfortable with it getting merged for reasons I can't fully explain (but are probably related to quality concerns)
